src/js/ui/menu.js

```diff
@@ -308,7 +308,7 @@ Menu.prototype.item = function(sectionIndex, itemIndex, item) {
 
 Menu.prototype.selection = function(callback_or_sectionIndex, itemIndex) {
   if (typeof callback_or_sectionIndex === 'function') {
-    this._selections.push(callback);
+    this._selections.push(callback_or_sectionIndex);
```

Comment:
Ah, yes, good point. I think what I was going for was separating variable jumping away from logic so that the two can be independently updated. It's more verbose, but I like the separation as it makes it more readable if you're aware the parameter shuffling is separated from logic.

To expand on my previous misleading snippet, it would then be:
```javascript
Menu.prototype.selection = function(sectionIndex, itemIndex) {
  var callback;
  if (typeof sectionIndex === 'function') {
    callback = sectionIndex;
    sectionIndex = undefined;
  }
  if (callback) {
    this._selections.push(callback);
    simply.impl.menuSelection();
  } else {
    this._selection = {
      sectionIndex: sectionIndex,
      itemIndex: itemIndex,
    };
    this._select();
  }
};
```

It's possible I didn't follow this style in some older parts of the code -- I should probably update those sections.